python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -556,6 +596,33 @@ def filter_out_invalid_p1_times(self, clear_existing: bool = False):
 
         return self
 
+    def get_available_source_ids(self, num_messages_to_read: int = 10):
+        self._populate_available_source_ids(num_messages_to_read=num_messages_to_read)
```

Comment:
This doesn't make any sense. `get_available_source_ids()` is meant to be a user-facing getter. It should just return `self.available_source_ids`.

`_populate_available_source_ids()` is totally separate and should be called by the constructor to populate the list one time and then not again.